onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -128,6 +128,48 @@ Status QnnBackendManager::LoadQnnSystemLib() {
   return Status::OK();
 }
 
+void QnnLogging(const char* format,
+                QnnLog_Level_t level,
+                uint64_t timestamp,
+                va_list argument_parameter) {
+  ORT_UNUSED_PARAMETER(level);
+  ORT_UNUSED_PARAMETER(timestamp);
+
+  // Always output Qnn log as Ort verbose log
+  size_t ini_size = 20;
+  std::vector<char> buffer(ini_size);
+  auto length = vsnprintf(buffer.data(), ini_size, format, argument_parameter);
```

Comment:
should we check for an error (negative return value)?

maybe consider using Capture::ProcessPrintf() which has error handling for various platforms:
https://github.com/microsoft/onnxruntime/blob/85cacf315ba77eb724649136951ec31a08ffc779/include/onnxruntime/core/common/logging/capture.h#L72

Reply:
that's a good idea. but need to do some changes to Capture code fist. Capture::ProcessPrintf doesn't provide the message, need to call Message() to get the message, and also need to create the Capture object, however ~Capture() will log the message.
Let me create another method on top of Capture::ProcessPrintf and return the string message directly.

---
In reply to: [1199507285](https://github.com/microsoft/onnxruntime/pull/16019#discussion_r1199507285) [](ancestors = 1199507285)

Reply:
updated.